### PR TITLE
Add `--format` flag to `lxd sql`

### PR DESCRIPTION
**Pull request by markylaing**
This allows specifying the output of `lxd sql` as `sql`, `csv`, `table`, `compact`, `json`, or `yaml`. The new `sql` format is added so that the column names are not auto-formatted (upper cased, replacing underscores with spaces).

This is added to ease scripting in tests and is required for #14651.

### CODE REVIEW

File: lxd/main_sql.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func sqlPrintSelectResult(format string, result internalSQLResult) error {
 	for _, row := range result.Rows {
 		r := make([]string, 0, len(row))
 		for _, col := range row {
-			r = append(r, fmt.Sprintf("%v", col))
+			r = append(r, fmt.Sprint(col))
```
Comment (tomponline) [inline lxd/main_sql.go:164]: i got hit by this bug today i think, so good spot.

File: shared/cmd/table.go [hunk 2]
```diff
@@ -35,6 +36,10 @@ func RenderTable(format string, header []string, data [][]string, raw any) error
 		table.SetHeaderLine(false)
 		table.SetBorder(false)
 		table.Render()
+	case TableFormatSQLResult:
+		table := getBaseTable(header, data)
+		table.SetAutoFormatHeaders(false)
```
Comment (tomponline) [inline shared/cmd/table.go:47]: why dont we want headers?

File: lxd/main_sql.go (hunk 1 above)
Comment (markylaing) [reply lxd/main_sql.go:164]: This is a bug? Also credit to @simondeziel for spotting this one!

File: shared/cmd/table.go (hunk 2 above)
Comment (markylaing) [reply shared/cmd/table.go:47]: This preserves the output format of `lxd sql` as it is. For example:
```
# Existing lxd sql format
$ lxd sql global 'select * from networks'
+----+------------+--------+-------------+-------+------+
| id | project_id |  name  | description | state | type |
+----+------------+--------+-------------+-------+------+
| 6  | 1          | lxdbr0 |             | 1     | 0    |
+----+------------+--------+-------------+-------+------+

# With auto formatted headers
$ lxd sql global 'select * from networks'
+----+------------+--------+-------------+-------+------+
| ID | PROJECT ID |  NAME  | DESCRIPTION | STATE | TYPE |
+----+------------+--------+-------------+-------+------+
| 6  | 1          | lxdbr0 |             | 1     | 0    |
+----+------------+--------+-------------+-------+------+
```

File: lxd/main_sql.go (hunk 1 above)
Comment (tomponline) [reply lxd/main_sql.go:164]: yeah i think so, or something similar as was  using `%` wildcards in my query and then i was seeing the unmatched parameter message.

File: shared/cmd/table.go (hunk 2 above)
Comment (tomponline) [reply shared/cmd/table.go:47]: cheers

**Review by tomponline — APPROVED** [review]

**Pull request merged**